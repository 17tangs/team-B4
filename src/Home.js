import React, {Component} from 'react';
import './Home.css';
import Card from './Card';
import SearchBar from './SearchBar';
import TopBar from './TopBar.js';
import axios from 'axios';
import {db, storageRef} from './fireApi';
import { createMuiTheme, ThemeProvider } from '@material-ui/core/styles';
import Button from "@material-ui/core/Button";



export class userData{
    constructor(obj){
        this.username    = obj.username;
        this.name        = obj.name;
        this.low         = obj.low;
        this.high        = obj.high;
        this.start_time  = obj.start_time;
        this.end_time    = obj.end_time;
        this.dist        = obj.dist;
        this.pic         = obj.pic;
        this.quote       = obj.quote;
        this.preferences = obj.preferences;
        this.promise     = null;
        this.email = obj.email ?? null;
    }

}
const _default = {username: '',
                 name: '',
                 low: 0,
                 high:0,
                 start_time: '',
                 end_time:'',
                 dist: 0,
                 quote: '',
                 pic:'',
                 preferences: []
             }



const theme = createMuiTheme({
    palette: {
        primary: {
            main: '#ffffff',
        },
        secondary: {
            main: '#2962ff',
        }
    },
    overrides: {
        MuiButton: {
            label: {
                color: '#ffffff',
            },
        }
    }
});

const style = {
    textTransform: 'none',
    borderRadius: '25px',
    width: '150px',
};


/**
 *    [WIP] Each added user will receive a Card component to represent them in a session.
 */
export default class Home extends Component{

    // Queries businesses around UCLA that match the specified category
    getLocation = category => {
      return axios.get(`${'https://cors-anywhere.herokuapp.com/'}https://api.yelp.com/v3/businesses/search`, {
        headers: {
          Authorization: `Bearer XyLNjPiVmPm-_-Og2rpIVSqVUNbsAihqwf21PVcmpbmhQow8HEAflaDDLiO8rT6SmehRVMyJNLz-OqjyiwXCqy45-EIE7yVttnY9440F04drNBm_ceiBgnsVUWNEXnYx`,
          'X-Requested-With': `XMLHttpRequest`,
        },
        params: {
          categories: `${category}`,

          // UCLA's coordinates
          latitude: 34.0689,
          longitude: -118.4452,
        }
      });
    };

    constructor(props){
        super(props);
        this.state={
            showSearch: false,
            all: [],
            users: [],
            filter: '',
            searchVal: '',
            showSearch: false,
            searchFocus: false,
            queryResult: null
        }
    }

    handleCardDelete = (card) => {
      var newAll = this.state.all.filter((x) => {return x !== card;});
      this.setState({all: newAll});
    }

    handleSearchBar = (e) => {
        if(e.key === 'Escape'){
            this.setState({
                showSearch: false,
                searchVal: ''
            })

        }
        if("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789".includes(e.key) &&
           !this.state.showSearch && this.state.searchVal === ''){
            this.setState({
                showSearch: true,
            })
        }

    }

    componentWillMount(){
        db.collection("users").get().then((querySnapshot) => {
                var l = [];
                querySnapshot.forEach((doc)=>{
                    l.push(new userData(doc.data()))});
                // this.setState({all: l})
                return l;
            }).then((users)=>{
                users.forEach((user)=>user.promise = this.getURL(user.pic));
                this.setState({all:users});
            })
        // var l = [];
        // this.state.all.forEach((user)=>{user.promise = this.getURL(user.pic); l.push(user)})
        // this.setState({all:l})
        document.addEventListener('keydown', this.handleSearchBar);

    }

    searchChange = (e) =>{
        this.setState({searchVal: e.target.value})
        // if(e.target.value===''){
        //     this.setState({
        //         showSearch:false
        //     })
        // }
    }
    getURL = (p) => storageRef.child(p).getDownloadURL();

    render(){

        // Sets state to first search result for default value "hiking"
        // Note it takes a few seconds to fetch this, but will fetch -> load new
        //    screen when displaying result
        if(this.state.queryResult == null) {
        // this.getLocation("hiking").then((response) =>
        //   this.setState({
        //     queryResult:response.data.businesses[0].name
        //   })
        // ).catch(function (response) {
        //   console.log(response);
        // });
      }
      console.log("query result: " + this.state.queryResult);
      console.log(this.state.all);
        return(
            <div >
            <div> <TopBar/> </div>
                <div className="flex justify-center" style={{paddingTop: 60}}>
                    {this.state.all.map((u)=><Card data={u} imgURL = {this.getURL(u.pic)} deleteCard = {this.handleCardDelete}/>)}
                </div>
                <div style={{marginTop: '5%', left: '45%', position: 'absolute'}} className="justify-center">
                    <ThemeProvider theme={theme}>
                        <div style={{display: 'flex', 'flex-direction': 'column'}}>
                            <Button variant={"contained"} onClick={()=>{this.handleSearchBar({key:''})}} theme={theme} color={"primary"} style={style}> <div style={{color: "grey"}}>  + Add Person </div> </Button>
                            <div style={{paddingTop: '20px'}}><Button variant={"contained"} onClick={()=>{/* TODO: make yelp query and change to suggestion page */}} theme={theme} color={"secondary"} style={style}> Find Me a Place! </Button></div>
                        </div>
                    </ThemeProvider>
                </div>
                <SearchBar userData={this.state.all} inputRef={this.inputRef} searchFocus={this.state.searchFocus} searchChange={this.searchChange} showSearch = {this.state.showSearch} searchVal ={this.state.searchVal}/>
            </div>
        );
    }


}
