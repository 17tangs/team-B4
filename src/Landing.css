
.Background-img {
  filter: blur(8px);
  -webkit-filter: blur(8px);
  min-height: 100%;
  min-width: 1024px;
  width: 100%;
  height: auto;
  position: fixed;
  top: 0;
  left: 0;
  z-index: -1;
}

.title {
  font-family: Raleway;
  color: #005DCB;
  font-weight: bold;
  font-size: 500%;

  margin: 0;
  position: absolute;
  top: 50%;
  left: 50%;
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
}
