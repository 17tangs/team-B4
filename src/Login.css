.login{
    display:flex;
    justify-content: center;
}

.login .box{
    font-size:20px;
    width:40%;
    box-shadow: 6px 6px 30px #CCC;
}
