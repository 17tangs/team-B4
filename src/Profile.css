/* want to have: .topnav, profile picture, login button*/

.profile{
    display:flex;
    justify-content: center;
    align-items: center;
    height:100%;
}

.topnav {
    height: 55px;
    width: 100%;
    background-color: #005DCB;
    display: block;
    
    margin-left: 0px;
    margin-right: 0px;

    position: absolute;
    /*
    top: 0;
    left: 0;
    */
}

.message {
    font-family: Raleway;
    cursor: pointer;
}

.gradient-box {
    border-radius: 20px;
    height: 500px;
    width: 300px;
    margin-left: 40px;
    background-image: linear-gradient(#e5f2ff, #FFFFFF);
}

.line {
    width: 100px;
    height: 1px;
    background-color: #000000;
}

.thumbnail-img {
    height: 50px;
    background-size: cover;
    display: block;
    border-radius: 100px;
    -webkit-border-radius: 100px;
    -moz-border-radius: 100px;

}

.fullname-display {
    font-family: Raleway;
    color: #37393B;
    /*color: #005DCB;*/
    font-weight: bold;
    font-size: 250%;
    position: absolute;
    margin-right: 400px;
    overflow: hidden;
    white-space: nowrap;
}

.username-display {
    font-family: Raleway;
    /*color: #80C1FF;*/
    color: #A3A6A7;
    position: absolute;
}

.username-display2 {
    font-family: Raleway;
    color: #005DCB;
    position: absolute;
    font-weight: bold;
}

.blurb-display {
    font-family: Raleway;
    position: absolute;
    overflow: hidden;
    white-space: nowrap;
}

.buttons {
    display: flex;
    flex-direction: column;
}

.profile-img {
    background-size: cover;
    display: block;
    height: 200px;
    border-radius: 300px;
    -webkit-border-radius: 300px;
    -moz-border-radius: 300px;
    /*margin-top: 200px;*/
}

.logistical-pref-display {
    display: inline;
    font-family: Raleway;
    overflow: hidden;
    white-space: nowrap;
}

.logistical-icon-img {
    display: inline;
    height: 25px;
    margin-right: 10px;
}

.row {
    display: flex;
    flex-direction: row;
}

.activity-img {
    background-size: cover;
    display: flex;
    flex-direction: row;
    height: 60px;
}

.edit-img {
    background-size: cover;
    display: flex;
    flex-direction: row;
    height: 30px;
}

.logout-img {
    background-size: cover;
    display: flex;
    flex-direction: row;
    cursor: pointer;
    height: 25px;
}

.tag-popup {
    font-family: Raleway;
    font-size: 80%;
    border-radius: 15px;
    background: #CFEBF9;
    padding: 5px;
    width: 60px;
    height: 25px;
    margin-left: 25%;
}

.label {
    display: inline;
    font-family: Raleway;
    font-size: 80%;
    border-radius: 15px;
    background: #CFEBF9;
    padding: 5px;
    width: 150px;
    height: 30px;
}

.dots {
    margin-top: 20px;
    margin-left: 10px;
    height: 30px;
    filter: invert(88%) sepia(19%) saturate(341%) hue-rotate(171deg) brightness(101%) contrast(95%);
}
