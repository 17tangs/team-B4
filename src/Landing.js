import React from "react";
import Button from "@material-ui/core/Button";
import "./Landing.css";
import backgr from './img/freeimg.jpg';
import { BrowserRouter, Switch, Route, Link } from "react-router-dom";
import {browserHistory} from 'react-router';

/**
 *   @return Landing page. Default page displayed when opening the app.
 */
class Landing extends React.Component {
  render() {
    return (
      <div>
        <img src={backgr} className="Background-img" alt=""/>
        <div  className="box-landing pa6 pt7">
          <div className="title"> Adventum </div>
            <div className="flex justify-center">
              <Button fullWidth style={{minWidth: '200px'}} color={"primary"} variant={"contained"} component={Link} to={{pathname:'/login',state:{login:true}}} >
                Log In
              </Button>
            </div>
            <div className="mv3 flex justify-center">
              <Button fullWidth style={{minWidth: '200px'}} variant={'contained'} component={Link} to={{pathname:'/login',state:{login:false}}} >
                Sign Up
              </Button>
            </div>
            <div className="mv3 flex justify-center" style={{fontSize:16, color:"#1D3756", 'font-family': "Raleway"}}>or</div>
            <div className="mv3 flex justify-center" style={{fontSize:20, textDecoration:'underline', color:"#888", 'font-family': "Raleway"}}>
                 <Link style={{color:"#1D3756"}} to="/">{'Continue as guest'}</Link>
            </div>
        </div>
      </div>

  );
  }
}


export default Landing;
