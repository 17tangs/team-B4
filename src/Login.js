import React from "react";
import './Login.css';
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import CircularProgress from "@material-ui/core/CircularProgress";
import { createMuiTheme, ThemeProvider } from '@material-ui/core/styles';
import Typography from "@material-ui/core/Typography";
import { withRouter } from 'react-router-dom';


const theme = createMuiTheme({
  overrides: {
    MuiInputBase: {
      root: {
        height: 50
      },
    },
  },
});

class Login extends React.Component {
  constructor(props) {
    super();
    this.state = {
      email: "",
      password: "",
      confirm_password: "",
      submitting: false,
      login: props.location.state.login,
    };
  }

  handleSubmit = e => {
    e.preventDefault();
    const { onSubmit } = this.props;
    const { email, password } = this.state;
    if (onSubmit) {
      this.setState({ submitting: true });
      onSubmit(email, password);
      this.setState({ submitting: false });
    }
  };

  handleSignIn = e => {
    e.preventDefault();
    const { onSignUp } = this.props;
    const { email, password } = this.state;
    if (onSignUp) {
      this.setState({ submitting: true });
      onSignUp(email, password);
      this.setState({ submitting: false });
    }
  };

  handleChange = key => e => {
    this.setState({ [key]: e.target.value });
  };


  render() {
    const { email, password, confirm_password, submitting } = this.state;
    console.log(this.props.error);
    return (
      <div className="login">
        <div className="box pa5">
            <Typography variant="h5" style={{ marginBottom: 24 }}>
              {this.state.login ? 'Login' : 'Signup' }
            </Typography>
            <form
              style={{ display: "flex", flexDirection: "column" }}
              onSubmit={this.state.login ? this.handleSubmit : this.handleSignIn}
            >
            <ThemeProvider theme={theme}>
              <TextField  style={{marginBottom: 24}} variant={"outlined"} required type={"email"} label={"Email"} value={email}
                onChange={this.handleChange("email")}
              />
              <TextField style={{marginBottom: 24}} variant={"outlined"} required type={"password"} label={"Password"} value={password}
                onChange={this.handleChange("password")}
              />
              {!this.state.login ?
              <TextField style={{marginBottom: 24}} variant={"outlined"} required type={"password"} label={"Confirm Password"} error={this.state.password != this.state.confirm_password && this.state.confirm_password != ''} helperText={this.state.password == this.state.confirm_password || this.state.confirm_password=='' ? "" : "Different from password"} value={confirm_password}
                onChange={this.handleChange("confirm_password")}
              /> : <div/>
               }
               </ThemeProvider>
               {this.props.error ? <div className="error mb3"> {this.props.error} </div> : <div/>}
              <Button
                type={"submit"}
                fullWidth
                variant={"contained"}
                color={"primary"}
              >
                {submitting ? (
                  <CircularProgress
                    style={{ color: "#fff" }}
                    color={"inherit"}
                    size={16}
                  />
              ) : (this.state.login ? "Login" : "Sign up"
                )}
              </Button>
              <div onClick={()=>{this.setState({'login': !this.state.login, 'email': '', 'password': '','confirm_password':''})}}> {this.state.login ? "Don't have an account? Sign up!" : "Already have an account? Log In!"} </div>
            </form>
            </div>
      </div>
    );
  }
}

export default withRouter(Login);
