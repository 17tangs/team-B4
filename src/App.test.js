import React from 'react';
import renderer from 'react-test-renderer';
import Landing from './Landing.js';
import Login from './Login.js';
import Home from './Home.js';
import Profile from './Profile.js';
import { MemoryRouter } from 'react-router-dom';

it('Landing renders correctly', () => {
  const tree = renderer
  	.create(
  		<MemoryRouter>
			<Landing />
		</MemoryRouter>
	).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Login renders correctly', () => {
	const tree = renderer
		.create(
			<MemoryRouter>
				<Login />
			</MemoryRouter>
		).toJSON();
	expect(tree).toMatchSnapshot();
});

it('Profile renders correctly', () => {
	const tree = renderer
		.create(
			<MemoryRouter>
				<Profile />
			</MemoryRouter>
		).toJSON();
	expect(tree).toMatchSnapshot();
});