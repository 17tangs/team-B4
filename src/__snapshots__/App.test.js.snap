// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly in "debug" mode 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
